cli/src/pazuzu_cli/console.clj

```diff
@@ -13,10 +13,40 @@
               any-url)]
     ((comp docker/generate-dockerfile registry/get-sorted-features) url features)))
 
-(defn save-docker-file
-  [docker-file-data  [& path]]
+(defn save-dockerfile
+  "Given Dockerfile data & path to save at (or defaults to Dockerfile<timestamp>),
+  creates the Dockerfile on disk. Returns the saved path."
+  [docker-file-data  & [ path]]
   (let [save-path (case path
                     nil (str "Dockerfile" (System/currentTimeMillis))
                     (str (clojure.string/join "" path) "/Dockerfile"))]
     (spit save-path docker-file-data)
     save-path))
+
+
+(def feature-flags {"-f" :features "-p" :path "--dry-run" :dry-run})
+
+
+(defn to-args-map
+  "Convert the list of args into a Hash Map charting all the arguments & flags.
```

Comment:
By the way, there is https://github.com/clojure/tools.cli
No need to reinvent the wheel :)